src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpChangeNamespaceService.cs

```diff
@@ -133,23 +133,34 @@ protected override SyntaxList<MemberDeclarationSyntax> GetMemberDeclarationsInCo
             // 4. When the namespace is specified and not "", i.e. we are moving referenced type to a different non-global 
             //    namespace. We need to replace the qualified reference with a new qualified reference (which is qualified 
             //    with new namespace.)
+            //
+            // Note that qualified type name can appear in QualifiedNameSyntax or MemberAccessSyntax, so we need to handle both cases.
 
             if (syntaxFacts.IsRightSideOfQualifiedName(nameRef))
             {
                 old = nameRef.Parent;
                 var aliasQualifier = GetAliasQualifierOpt(old);
 
-                if (IsGlobalNamespace(newNamespaceParts))
+                if (!TryGetGlobalQualifiedName(newNamespaceParts, nameRef, aliasQualifier, out @new))
                 {
-                    // If new namespace is "", then name will be declared in global namespace.
-                    // We will replace qualified reference with simple name qualified with alias (global if it's not alias qualified)
-                    var aliasNode = aliasQualifier?.ToIdentifierName() ?? SyntaxFactory.IdentifierName(SyntaxFactory.Token(SyntaxKind.GlobalKeyword));
-                    @new = SyntaxFactory.AliasQualifiedName(aliasNode, nameRef.WithoutTrivia());
+                    var qualifiedNamespaceName = CreateNamespaceAsQualifiedName(newNamespaceParts, aliasQualifier, newNamespaceParts.Length - 1);
+                    @new = SyntaxFactory.QualifiedName(qualifiedNamespaceName, nameRef.WithoutTrivia());
                 }
-                else
+
+                // We might lose some trivia associated with children of `outerMostNode`.  
+                @new = @new.WithTriviaFrom(old);
+                return true;
+            }
+
+            if (syntaxFacts.IsNameOfMemberAccessExpression(nameRef))
```

Comment:
Nit: else if, since it doesn't require you to visually scan that the previous if will always return.